cmd/gateway-azure.go

```diff
@@ -705,31 +767,25 @@ func (a *azureObjects) CompleteMultipartUpload(bucket, object, uploadID string,
 		}
 	}
 
-	err = a.client.PutBlockList(bucket, object, allBlocks)
+	err = objBlob.PutBlockList(allBlocks, nil)
 	if err != nil {
 		return objInfo, azureToObjectError(traceError(err), bucket, object)
 	}
-	if len(meta) > 0 {
-		prop := storage.BlobHeaders{
-			ContentMD5:      meta["Content-Md5"],
-			ContentLanguage: meta["Content-Language"],
-			ContentEncoding: meta["Content-Encoding"],
-			ContentType:     meta["Content-Type"],
-			CacheControl:    meta["Cache-Control"],
-		}
-		err = a.client.SetBlobProperties(bucket, object, prop)
+	if len(metadata.Metadata) > 0 {
+		objBlob.Metadata, objBlob.Properties = s3MetaToAzure(metadata.Metadata)
+		err = objBlob.SetProperties(nil)
 		if err != nil {
 			return objInfo, azureToObjectError(traceError(err), bucket, object)
 		}
-		err = a.client.SetBlobMetadata(bucket, object, nil, meta)
+		err = objBlob.SetMetadata(nil)
 		if err != nil {
 			return objInfo, azureToObjectError(traceError(err), bucket, object)
 		}
 	}
 	return a.GetObjectInfo(bucket, object)
 }
 
-// Copied from github.com/Azure/azure-sdk-for-go/storage/blob.go
+// Copied from github.com/Azure/azure-sdk-for-go/storage/container.go
 func azureListBlobsGetParameters(p storage.ListBlobsParameters) url.Values {
```

Comment:
As this function is used in `gateway-azure-anonymous.go`, you could move it there.

Reply:
Done.